### PR TITLE
Fail az login only if az CLI exits with code != 0

### PR DESCRIPTION
**Pull request by doubleyewdee**
We should not fail the task on az login / account set even if there is output to stderr (this isn't a reliable indicator of failure). Fail only if the actual command run exits with a non-zero status code which will appropriately indicate failure.

### CODE REVIEW

**Comment by bryanmacfarlane** [comment]
@paulgmiller - yeah it's the "other" stream :) . many tools alert, status on stderr so obviously should use return code.  Can you log an issue?

File: Tasks/AzureCLIV1/azureclitask.ts [hunk 1]
```diff
@@ -140,7 +140,7 @@ export class azureclitask {
     }
 
     private static throwIfError(resultOfToolExecution): void {
-        if (resultOfToolExecution.stderr) {
+        if (resultOfToolExecution.code != 0) {
```
Comment (bryanmacfarlane) [inline Tasks/AzureCLIV1/azureclitask.ts]: Perhaps the function should be throwIfFailed to make it clearer?

Comment (doubleyewdee) [reply Tasks/AzureCLIV1/azureclitask.ts]: Good call. Fixed.

**Comment by sachinma** [comment]
@vineetmimrot missed this PR. He raised a separate PR and has merged it - I am closing this one.

**Pull request closed without merging**